src/box/box.cc

```diff
@@ -782,28 +764,41 @@ box_check_election_timeout(void)
 	return d;
 }
 
-static void
-box_check_replication(void)
+static int
+box_check_uri_set(const char *option_name)
 {
-	int count = cfg_getarr_size("replication");
-	for (int i = 0; i < count; i++) {
-		const char *source = cfg_getarr_elem("replication", i);
-		if (box_check_uri(source, "replication") != 0)
-			diag_raise();
+	struct uri_set uri_set;
+	if (cfg_get_uri_set(option_name, &uri_set) != 0) {
+		diag_set(ClientError, ER_CFG, option_name,
+			 diag_last_error(diag_get())->errmsg);
+		return -1;
 	}
+	int rc = 0;
+	for (int i = 0; i < uri_set.uri_count && rc == 0; i++) {
+		const struct uri *uri = &uri_set.uris[i];
+		if (uri->service == NULL) {
+			char *uristr = tt_static_buf();
+			uri_format(uristr, TT_STATIC_BUF_LEN, uri, false);
+			diag_set(ClientError, ER_CFG, option_name,
+				 tt_sprintf("bad URI '%s': expected host:service "
+				 "or /unix.socket", uristr));
+			rc = -1;
```

Comment:
You can break from the loop here. Otherwise you will overwrite diag with same multiple errors in case there is more than one.

Reply:
it's break because rc == -1 and condition of the loop rc == 0.

Reply:
Unconditional jump (break or goto) is better than additional test.

Reply:
fixed